app/models/repository.rb

```diff
@@ -61,12 +61,14 @@ def self.add_repo(event, namespace, repo, tag)
     if repository.nil?
       repository = Repository.create(namespace: namespace, name: repo)
     elsif repository.tags.exists?(name: tag)
-      return
+      repository.tags.where(name: tag).each { |tag| tag.touch }
```

Comment:
return isn't needed here. maybe even an improvement, because now this method returns always a repository